src/Constructorio_NET.Tests/client/modules/FacetOptionTest.cs

```diff
@@ -0,0 +1,264 @@
+            newFacetOptions[0].Position = null;
+            newFacetOptions[0].Data = null;
+
+            newFacetOptions[1].DisplayName = "1";
+            newFacetOptions[1].Position = -1;
+            newFacetOptions[1].Data = new System.Collections.Hashtable();
+
+            List<FacetOption> updatedFacets = await constructorio.Catalog.BatchCreateOrUpdateFacetOptions(newFacetOptions, this.facetGroupName);
+            FacetOption facetOption0 = updatedFacets.Find((f) => f.Value == newFacetOptions[0].Value);
+            FacetOption facetOption1 = updatedFacets.Find((f) => f.Value == newFacetOptions[1].Value);
+
+            Assert.AreEqual("0", facetOption0.DisplayName, "Updated Facet has incorrectly updated Display Name");
+            Assert.AreEqual(1, facetOption0.Position, "Updated Facet incorrectly updated Position");
+            Assert.AreEqual(data, facetOption0.Data, "Updated Facet incorrectly updated Data");
+            Assert.AreEqual("1", facetOption1.DisplayName, "Updated Facet has incorrectly updated Display Name");
+            Assert.AreEqual(null, facetOption1.Position, "Updated Facet has incorrectly updated Position");
+            Assert.AreEqual(new System.Collections.Hashtable(), facetOption1.Data, "Updated Facet has incorrectly updated Data");
+        }
+
+        [Test]
+        public async Task GetFacetOption()
+        {
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            FacetOption facetOption = CreateRandomFacetOption();
+            await constructorio.Catalog.CreateFacetOption(facetOption, this.facetGroupName);
+
+            FacetOption fetchedFacetOption = await constructorio.Catalog.GetFacetOption(facetOption.Value, this.facetGroupName);
+
+            Assert.AreEqual(facetOption.DisplayName, fetchedFacetOption.DisplayName, "Fetched facet has different Display Name");
+            Assert.AreEqual(facetOption.Value, fetchedFacetOption.Value, "Fetched facet option has different Value");
+            Assert.AreEqual(facetOption.Hidden, fetchedFacetOption.Hidden, "Fetched facet option has different Hidden value");
+        }
+
+        [Test]
+        public async Task ReplaceFacetOption()
+        {
+            ConstructorIO constructorio = new ConstructorIO(this.Config);
+            FacetOption facetOption = CreateRandomFacetOption();
+            await constructorio.Catalog.CreateFacetOption(facetOption, this.facetGroupName);
+
+            facetOption.DisplayName = "updated";
+            facetOption.Position = 2;
+            facetOption.Hidden = false;
+            facetOption.Data = new System.Collections.Hashtable();
+            facetOption.Data.Add("hello", 3);
+
+            FacetOption replacedFacetOption = await constructorio.Catalog.ReplaceFacetOption(facetOption, this.facetGroupName);
+
+            Assert.AreEqual(facetOption.Value, replacedFacetOption.Value, "Replaced facet option has non-updated Value");
```

Comment:
I don't think this assertion is needed since the value shouldn't change.

Reply:
You're absolutely right, removed this line.